### PR TITLE
Fix #312, Generate Documentation In All Scenarios

### LINKED ISSUE #312: Generate documentation PDF in all scenarios regardless of whether it will be pushed or not.
## Problem

We could have caught the [latest error](https://github.com/nasa/cFS/runs/3107101587) in the documentation build and push job if we built the PDFs every time. 

## Proposed Solution 

Only keep this condition in the "push" step of the job; remove it from the dependencies and document generation steps.

```yml
if: ${{ github.event_name == 'push' && contains(github.ref, 'main')}}
```

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Checklist (Please check before submitting)**

* [X] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [ ] I signed and emailed the appropriate Contributor License agreement to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Fixes #312

remove `if: ${{ github.event_name == 'push' && contains(github.ref, 'main')}}` from PDF generation installs and PDF generation. Add PDF generation installs and PDF generation steps to the previous job, so that Deploy is a separate job. 

**Testing performed**
N/A

**Expected behavior changes**
PDFs will generate no matter what while deployment is based on if it is pushed on main. 

**Additional context**
May require cache PR for successful deployment: https://github.com/nasa/cFS/pull/315

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

File: .github/workflows/build-documentation.yml [hunk 1]
```diff
@@ -167,6 +167,21 @@ jobs:
           # Could add pandoc and convert to github markdown
           # pandoc CFE_Users_Guide.pdf -t gfm
 
+  deploy-usersguide:
```
Comment (astrogeco) [inline .github/workflows/build-documentation.yml:170]: Not sure if this is a merge artifact but there's two "deploy" jobs per guide now. See line 139 and 257. I think what we want is to move the steps from the jobs defined by 139 and 257 and move them into the previous job.

File: .github/workflows/build-documentation.yml [hunk 2]
```diff
@@ -167,6 +167,21 @@ jobs:
...
+    needs: build-usersguide
+    # Name the Job
+    name: Deploy Users Guide
+    # Set the type of machine to run on
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: Cache cFS Build Environment for usersguide
+        id: cache-bundle
```
Comment (astrogeco) [inline .github/workflows/build-documentation.yml:179]: should come up with a unique name for each cache step, maybe `cache_usersguide` and `cache_osalguide`?

**Review by astrogeco — CHANGES_REQUESTED** [review]

File: .github/workflows/build-documentation.yml (hunk 1 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/build-documentation.yml:155]: Resolved

File: .github/workflows/build-documentation.yml (hunk 2 above)
Comment (ArielSAdamsNASA) [reply .github/workflows/build-documentation.yml]: Used cache_usersguide and cache_osalguide

**Review by astrogeco — APPROVED** [review]

**Pull request merged**